src/view/switch.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+
+use crate::view::ViewMarker;
+use crate::{view::Id, widget::ChangeFlags, MessageResult};
+
+use super::{Cx, View};
+
+pub struct Switch {
+    is_on: bool,
+}
+
+pub fn switch(is_on: bool) -> Switch {
```

Comment:
I think this should probably have a callback parameter as well, for when the value is changed, similar as in the button example, but since there's a payload (`bool`), it should probably be an `impl Fn(&mut T, bool) -> A` or something like that.

So that it looks as something like this:

```rust
fn my_view(state: &mut AppState) -> impl View<AppState> {
    switch(state.my_switch_value, |state, value| { state.my_switch_value = value; })
}
```

But it would be interesting to "sugar" this callback boilerplate away somehow,
as I think something like the callback above is probably the main use-case, though I think often something additionally should happen when the value is changed as well.

For the simple case above that can probably be done via a macro? But I guess something like that should probably be done on top of this PR in a different PR...
The "UX design" of views is still a relatively big question mark so far (needs more experimentation and discussion to find the most idiomatic style IMO)
(I think SwiftUI is probably the best source of inspiration for most of this stuff)